### ISSUE TITLE
Serving docs may result in sticking "version selector", when arbitrary title is used

### ISSUE THREAD

**Issue by neatc0der** [zayd62/mkdocs-versioning#61]
**Is your feature request related to a problem? Please describe.**
It's related to #59.

The current title for "version selector" is hardcoded.

**Describe the solution you'd like**
Replace this inadequacy with an additional configuration item.

**Describe alternatives you've considered**
There are none.

**Additional context**
This feature is independent of external factors.

**Comment by zayd62** [comment]
thanks for the suggestion and for the implementation in #62 

I'll merge it into a separate branch along with #60 (already done), write some additional docs and then merge into master where it will then be available via pip

**Comment by zayd62** [comment]
hey so im not too sure what the option is supposed to change as im not seeing any changes with the option mentioned and the build fails if you don't specify the option (i'll fix the error), perhaps if you could show me a screenshot of what should happen if the option is specified so i know what the output is supposed to look like

**Comment by neatc0der** [comment]
Right, so there's probably a `KeyError` when executing `mkversion/entry.py:73` without the configuration parameter `version_selector_title` set in `mkdocs.yml`.

What it's meant to do is simple: Adjust the title for "Version Selector" to an arbitrary string, like "go back to version overview".

I've added a comment on my [original commit](https://github.com/neatc0der/mkdocs-versioning/commit/69c93d5dff171511a1f404bddb6954440e8bf576#diff-c7e9607b82010759499e47b73af2b3cf456f46d55cf69f69e5e38aa2098d98ccR73) to show the bug fix.

Sorry for the inconvenience. This merely happened due to purely executed testing.

**Comment by zayd62** [comment]
![image](https://user-images.githubusercontent.com/26032382/105641552-719cf700-5e7c-11eb-8bcc-a87863b3d11f.png)

Do you mean this?

**Comment by neatc0der** [comment]
Yes.

**Comment by zayd62** [comment]
so that is to do with the `mkdocs.yml`. using the plugin docs as an example, https://zayd62.github.io/mkdocs-versioning/0.3.0/ and the `mkdocs.yml` below

```
nav:
  - Home: "index.md"
  - Detail:
      - How it works: "reference/index.md"
      - Step by Step Example: "reference/step_by_step.md"
      - Config Options: "reference/config_options.md"
      - CLI Options: "reference/cli.md"
  - Version Selector: "../"
  ```

you can see on the docs that `Home` and `Version Selector` appears on the left and `Detail` appears as a tab which you can click on and will take you to a page with all the other pages (` How it works`, etc)

currently, the implementation to check if the version selection nav item is included requires the phrase `Version Selector` to exists. fix would be to check if the value `../` exists in the nav that way you can name it whatever you want so this config option isn't really needed

**Comment by neatc0der** [comment]
Gotcha, but keep in mind [mkversion/entry.py:47](https://github.com/zayd62/mkdocs-versioning/blob/387ccd966289bf142fd45df94110a43766867a10/mkversion/entry.py#L47).

Without being able to configure this name, the main page linking to all versions will also have a link like "go back to version overview" if you derive from "Version Selector" as title. Thus this is actually a bug fix :smile:

**Comment by zayd62** [comment]
https://zayd62.github.io/mkdocs-versioning/ the version selection doesnt have that option. just did a local build and it did not show up either 
![image](https://user-images.githubusercontent.com/26032382/105644132-c8f69380-5e8b-11eb-8f2f-f522c78eb2db.png)
 
perhaps if you could provide me with some docs which i can use to reproduce the error

**Comment by neatc0der** [comment]
Ok, I'm finally getting this line of code...

When using `mkdocs serve`, the version selector is supposed to get filtered from the menu. But if you call it differently... Well, it sticks, which is not that big of a deal.

Sorry for holding you up with this, _I'll finally withdraw_.

Conclusion: It's actually a minor bug, which would not be satisfactory fixed by the provided merge request. :neutral_face:

**Comment by zayd62** [comment]
![image](https://user-images.githubusercontent.com/26032382/105644719-f98bfc80-5e8e-11eb-8aa2-391a4bd89a99.png)

i see what you mean. the whole point of serving is to see what the page looks like without having to build. part of the page is the link which takes you to the version selection page so i decided to keep that in although i can see that it can be misleading as `mkdocs serve` does not serve the version selection page, only the version that you are working on. The reason for this behaviour is because the version selection page is generated after mkdocs builds the latest version and with how `mkdocs serve` works internally

i decided not to serve the built docs with the version selector as initially, you could not specify a custom version selection page. Now that you can specify a custom version selection page, perhaps `mkdocs serve` should show you the version selection page as well.

i feel that the plugin is feature complete so apart from unit tests and cleaning up the code (functions, documentation etc) i think the plugin is ready for a 1.0 release

anyways, ive merged #60 and wrote a fix so **version selector** is not hard coded in. its available in the `neatc0der` branch so if you could

1. see if awesome pages now works after the merge
2. see if you can rename the version selector text to whatever you want

if its all good, ill merge to master and then the plugin will be uploaded to pypi if i get a chance in the weekday evening otherwise weekend

**Comment by neatc0der** [comment]
Ok, so there's #63. After that, it should be fine. Thanks for your time and consideration!
